spring-kafka/src/main/java/org/springframework/kafka/listener/AbstractMessageListenerContainer.java

```diff
@@ -239,6 +256,123 @@ public void setAutoStartup(boolean autoStartup) {
+		this.errorHandler = errorHandler;
+	}
+
+	protected Executor getConsumerTaskExecutor() {
+		return this.consumerTaskExecutor;
+	}
+
+	/**
+	 * Set the executor for threads that poll the consumer.
+	 * @param consumerTaskExecutor the executor
+	 */
+	public void setConsumerTaskExecutor(Executor consumerTaskExecutor) {
+		this.consumerTaskExecutor = consumerTaskExecutor;
+	}
+
+	protected Executor getListenerTaskExecutor() {
+		return this.listenerTaskExecutor;
+	}
+
+	/**
+	 * Set the executor for threads that invoke the listener.
+	 * @param listenerTaskExecutor the executor.
+	 */
+	public void setListenerTaskExecutor(Executor listenerTaskExecutor) {
+		this.listenerTaskExecutor = listenerTaskExecutor;
+	}
+
+	protected long getPauseAfter() {
+		return this.pauseAfter;
+	}
+
+	/**
+	 * When using Kafka group management and {@link #setPauseEnabled(boolean)} is
+	 * true, the delay after which the consumer should be paused. Default 10000.
+	 * @param pauseAfter the delay.
+	 */
+	public void setPauseAfter(long pauseAfter) {
+		this.pauseAfter = pauseAfter;
+	}
+
+	protected boolean isPauseEnabled() {
+		return this.pauseEnabled;
+	}
+
+	/**
+	 * Set to true to avoid rebalancing when this consumer is slow or throws a
+	 * qualifying exception - pause the consumer.
+	 * Default: true.
+	 * @param pauseEnabled true to pause.
+	 * @see #setPauseAfter(long)
+	 */
+	public void setPauseEnabled(boolean pauseEnabled) {
+		this.pauseEnabled = pauseEnabled;
+	}
+
+	protected RetryTemplate getRetryTemplate() {
+		return this.retryTemplate;
+	}
+
+	/**
+	 * Set a retry template to retry deliveries.
+	 * @param retryTemplate the retry template.
+	 */
+	public void setRetryTemplate(RetryTemplate retryTemplate) {
+		this.retryTemplate = retryTemplate;
+	}
+
+	protected RecoveryCallback<Void> getRecoveryCallback() {
+		return this.recoveryCallback != null ? this.recoveryCallback :
+			new RecoveryCallback<Void>() {
+
+				@Override
+				public Void recover(RetryContext context) throws Exception {
+					@SuppressWarnings("unchecked")
+					ConsumerRecord<K, V> record = (ConsumerRecord<K, V>) context.getAttribute("record");
+					Throwable lastThrowable = context.getLastThrowable();
+					if (getErrorHandler() != null && lastThrowable instanceof Exception) {
+						getErrorHandler().handle((Exception) lastThrowable, record);
+					}
+					else {
+						AbstractMessageListenerContainer.this.logger
+							.error("Listener threw an exception and no error handler for " + record, lastThrowable);
+					}
+					return null;
+				}
+
+			};
+	}
+
+	/**
+	 * Set a recovery callback to be invoked when retries are exhausted. By
+	 * default the error handler is invoked.
+	 * @param recoveryCallback the recovery callback.
+	 */
+	public void setRecoveryCallback(RecoveryCallback<Void> recoveryCallback) {
+		this.recoveryCallback = recoveryCallback;
+	}
+
+
+	protected int getQueueDepth() {
+		return this.queueDepth > 0 ? this.queueDepth : 1;
```

Comment:
You say in the next JavaDocs that it is `Default 2`, but return `1`.
Is that intentional or omission ?